events/planner.py

```diff
@@ -0,0 +1,45 @@
+from events.models import OptionalMeetingDates, PossibleParticipant, Event
+
+
+class EventPlanner():
+
+    def find_meeting(self, event_id):
+        """ Find a meeting date with all the participants if possible.
+            The planner will look over all the possible dates and
+            will check if there is a possible date that works for everyone
+        """
+        results = list()
+        all_participants = PossibleParticipant.objects.get_all_possible_participants_amount(event_id)
+        for possible_date in OptionalMeetingDates.objects.get_all_event_dates(event_id):
+            participants_amount = PossibleParticipant.objects.get_all_date_participants(possible_date).count()
+            if all_participants == participants_amount:
+                results.append(possible_date)
```

Comment:
AFAIU the "run" method only take actions on the first possible_date found, so is it necessary to continue the loop after possible_date is found?

isn't returning the first possible_date that match the if statment sufficient?

Reply:
@RonHachmon Actually with that we can (If wish in future) to let the user choose between a few possible dates

Reply:
Yes, that is currently what is happening. I thought it would be possible to implement this later with a random selection of a date from among the possible dates that the algorithm finds or to show all the possibilities to the user and let him choose, so I implemented it this way.
But at the moment, it seems to me that we will stay with the selection of the first date that fits as a returned value of the algorithm, so I will change the way it is implemented.